src/topology/metric_space/thickened_indicator.lean

```diff
@@ -0,0 +1,211 @@
+and `0` outside a `δ`-thickening of `E` and interpolates (continuously) between
+these values using `inf_edist _ E`.
+
+`thickened_indicator_aux` is the unbundled `ℝ≥0∞`-valued function. See `thickened_indicator`
+for the (bundled) bounded continuous function with `ℝ≥0`-values. -/
+def thickened_indicator_aux (δ : ℝ) (E : set α) : α → ℝ≥0∞ :=
+λ (x : α), (1 : ℝ≥0∞) - (inf_edist x E) / (ennreal.of_real δ)
+
+lemma continuous_thickened_indicator_aux {δ : ℝ} (δ_pos : 0 < δ) (E : set α) :
+  continuous (thickened_indicator_aux δ E) :=
+begin
+  unfold thickened_indicator_aux,
+  let f := λ (x : α), (⟨1, (inf_edist x E) / (ennreal.of_real δ)⟩ : ℝ≥0 × ℝ≥0∞),
+  let sub := λ (p : ℝ≥0 × ℝ≥0∞), ((p.1 : ℝ≥0∞) - p.2),
+  rw (show (λ (x : α), ((1 : ℝ≥0∞)) - (inf_edist x E) / (ennreal.of_real δ)) = sub ∘ f, by refl),
+  apply (@ennreal.continuous_nnreal_sub 1).comp,
+  apply (ennreal.continuous_div_const (ennreal.of_real δ) _).comp continuous_inf_edist,
+  norm_num [δ_pos],
+end
+
+lemma thickened_indicator_aux_le_one (δ : ℝ) (E : set α) (x : α) :
+  thickened_indicator_aux δ E x ≤ 1 :=
+by apply @tsub_le_self _ _ _ _ (1 : ℝ≥0∞)
+
+lemma thickened_indicator_aux_lt_top {δ : ℝ} {E : set α} {x : α} :
+  thickened_indicator_aux δ E x < ∞ :=
+lt_of_le_of_lt (thickened_indicator_aux_le_one _ _ _) one_lt_top
+
+lemma thickened_indicator_aux_closure_eq (δ : ℝ) (E : set α) :
+  thickened_indicator_aux δ (closure E) = thickened_indicator_aux δ E :=
+by simp_rw [thickened_indicator_aux, inf_edist_closure]
+
+lemma thickened_indicator_aux_one (δ : ℝ) (E : set α) {x : α} (x_in_E : x ∈ E) :
+  thickened_indicator_aux δ E x = 1 :=
+by simp [thickened_indicator_aux, inf_edist_zero_of_mem x_in_E, tsub_zero]
+
+lemma thickened_indicator_aux_one_of_mem_closure
+  (δ : ℝ) (E : set α) {x : α} (x_mem : x ∈ closure E) :
+  thickened_indicator_aux δ E x = 1 :=
+by rw [←thickened_indicator_aux_closure_eq, thickened_indicator_aux_one δ (closure E) x_mem]
+
+lemma thickened_indicator_aux_zero
+  {δ : ℝ} (δ_pos : 0 < δ) (E : set α) {x : α} (x_out : x ∉ thickening δ E) :
+  thickened_indicator_aux δ E x = 0 :=
+begin
+  rw [thickening, mem_set_of_eq, not_lt] at x_out,
+  unfold thickened_indicator_aux,
+  apply le_antisymm _ bot_le,
+  have key := tsub_le_tsub (@rfl _ (1 : ℝ≥0∞)).le (ennreal.div_le_div x_out rfl.le),
+  rw [ennreal.div_self (ne_of_gt (ennreal.of_real_pos.mpr δ_pos)) of_real_ne_top] at key,
+  simpa using key,
+end
+
+lemma thickened_indicator_aux_mono {δ₁ δ₂ : ℝ} (hle : δ₁ ≤ δ₂) (E : set α) :
+  thickened_indicator_aux δ₁ E ≤ thickened_indicator_aux δ₂ E :=
+λ _, tsub_le_tsub (@rfl ℝ≥0∞ 1).le (ennreal.div_le_div rfl.le (of_real_le_of_real hle))
+
+lemma thickened_indicator_aux_subset (δ : ℝ) {E₁ E₂ : set α} (subset : E₁ ⊆ E₂) :
+  thickened_indicator_aux δ E₁ ≤ thickened_indicator_aux δ E₂ :=
+λ _, tsub_le_tsub (@rfl ℝ≥0∞ 1).le
+  (ennreal.div_le_div (inf_edist_le_inf_edist_of_subset subset) rfl.le)
+
+lemma thickened_indicator_aux_tendsto_indicator_closure
```

Comment:
maybe add a docstring to this lemma, reminding us that the topology here is the topology of pointwise convergence.